scheduler/mesos/mesos.go

```diff
@@ -222,56 +235,71 @@ func (m *mesosScheduler) getMesosMaster() ([]string, string, error) {
 	return masterHosts, protocol, nil
 }
 
-func (m *mesosScheduler) getMesosTask(taskId string) (mesosTask, string, string, error) {
-	var state mesosState
-	var masterErr error
-	if masterHosts, protocol, err := m.getMesosMaster(); err == nil {
-		for _, host := range masterHosts {
-			if resp, err := http.Get(protocol + "://" + host + "/state.json"); err == nil {
-				defer resp.Body.Close()
-				if err := json.NewDecoder(resp.Body).Decode(&state); err == nil {
-					if state.Pid == state.Leader {
```

Comment:
Do we need to make sure we are on the leader when we get this info?

Reply:
No. The API redirects you to the leading master automatically.